src/Infrastructure/LeanCode.AuditLogs/AuditLogsPublisher.cs

```diff
@@ -0,0 +1,44 @@
+using System.Diagnostics;
+using LeanCode.OpenTelemetry;
+using LeanCode.TimeProvider;
+using MassTransit;
+using Microsoft.EntityFrameworkCore;
+
+namespace LeanCode.AuditLogs;
+
+public static class AuditLogsPublisher
```

Comment:
Let's make this a proper, non-static class. Will be easier to unit-test the filter & middleware (and will follow OOP principles more).